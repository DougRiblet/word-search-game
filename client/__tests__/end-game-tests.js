import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Utils from 'react-dom/test-utils'; 
import EndButton from '../components/End-Game/end-game-present';

describe('End Game', () => {
  test('Snapshot', () => {
    const EndButton_tree = renderer.create(
      <EndButton endGame={() => endGame()} allownew={false} />
    ).toJSON();
    expect(EndButton_tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<EndButton endGame={() => endGame()} allownew={false} />, div);
  });

  test('calls endGame when button is clicked', () => {
    const endGame = jest.fn();
    const wrapper = mount(
      <EndButton endGame={endGame} allownew={false} />
    );
    const butt = wrapper.find('#end-game-button');
    butt.simulate('click');
    expect(endGame).toHaveBeenCalled();
  });

});
