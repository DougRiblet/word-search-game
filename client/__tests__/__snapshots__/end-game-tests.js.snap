// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End Game Snapshot 1`] = `
<div
  id="endgame"
>
  <button
    disabled={false}
    id="end-game-button"
    onClick={[Function]}
  >
    End Game
  </button>
</div>
`;
